src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c

```diff
@@ -0,0 +1,806 @@
+        jint jwidth,
+        jint jheight,
+        jintArray pixelArray,
+        jstring jtoken
+) {
+
+    GdkRectangle requestedArea = { jx, jy, jwidth, jheight};
+
+    const gchar *token = jtoken
+                         ? token = (*env)->GetStringUTFChars(env, jtoken, NULL)
+                         : NULL;
+
+    DEBUG_SCREENCAST(
+            "taking screenshot at \n\tx: %5i y %5i w %5i h %5i token |%s|\n",
+            jx, jy, jwidth, jheight, token
+    );
+
+    if (!initScreencast(token)) {
+        releaseToken(env, jtoken, token);
+        if (pw.pwFd == START_DENIED) {
+            return START_DENIED;
+        } else {
+            return -1;
+        }
+    }
+    if (!doLoop(requestedArea)) {
+        releaseToken(env, jtoken, token);
+        return -1;
+    }
+
+    while (!isAllDataReady()) {
+        fp_pw_thread_loop_wait(pw.loop);
+    }
+
+    DEBUG_SCREENCAST("\nall data ready\n", NULL);
+
+    for (int i = 0; i < screenSpace.screenCount; ++i) {
+        struct ScreenProps * screenProps = &screenSpace.screens[i];
+
+        if (screenProps->shouldCapture) {
+            GdkRectangle bounds = screenProps->bounds;
+            GdkRectangle captureArea  = screenProps->captureArea;
+            DEBUG_SCREEN_PREFIX(screenProps,
+                                "@@@ copying screen data %i, captureData %p\n"
+                                "\t||\tx %5i y %5i w %5i h %5i %s\n"
+                                "\t||\tx %5i y %5i w %5i h %5i %s\n"
+                                "\t||\tx %5i y %5i w %5i h %5i %s\n\n",
+                                i, screenProps->captureData,
+                                requestedArea.x, requestedArea.y,
+                                requestedArea.width, requestedArea.height,
+                                "requested area",
+
+                                bounds.x, bounds.y,
+                                bounds.width, bounds.height,
+                                "screen bound",
+
+                                captureArea.x, captureArea.y,
+                                captureArea.width, captureArea.height,
+                                "in-screen coords capture area"
+            );
+
+            if (screenProps->captureData) {
+                for (int y = 0; y < captureArea.height; y++) {
+                    jsize preY = (requestedArea.y > screenProps->bounds.y)
+                            ? 0
+                            : screenProps->bounds.y - requestedArea.y;
+                    jsize preX = (requestedArea.x > screenProps->bounds.x)
+                            ? 0
+                            : screenProps->bounds.x - requestedArea.x;
+                    jsize start = jwidth * (preY + y) + preX;
+
+                    jsize len = captureArea.width;
+
+                    (*env)->SetIntArrayRegion(
+                            env, pixelArray,
+                            start, len,
+                            ((jint *) screenProps->captureData)
```

Comment:
I'm not sure if endianness of `captureData` matches the expected endianness of `pixelArray`.

Reply:
For now, it is going to be supported on x86-64 only, but it is a good candidate for investigation.

Reply:
Is this enforced somehow? I mean what happens if this code is built on risc-v or arm?